pkgs/by-name/ar/argon/package.nix

```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  udev,
+  zstd,
+  stdenv,
+  darwin,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "argon";
+  version = "2.0.20";
+
+  src = fetchFromGitHub {
+    owner = "argon-rbx";
+    repo = "argon";
+    rev = version;
+    hash = "sha256-tScadRUKKS+n4d4fIv5h32W63R1a6XWx5olO5/KmoS8=";
+  };
+
+  cargoHash = "sha256-DGD1e3WoWA33+xnQaGzLO4AN/EfcinW+rGEkUTWo/lA=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs =
+    [
+      udev
+      zstd
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      darwin.apple_sdk.frameworks.CoreFoundation
+      darwin.apple_sdk.frameworks.CoreServices
+      darwin.apple_sdk.frameworks.Foundation
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.SystemConfiguration
```

Comment:
```suggestion
```
Since my last review, the Darwin SDKs were updated to no longer require this, I think.

Reply:
The package didn't build on my M2 system without at least having Foundation and SystemConfiguration.

Reply:
You may need to rebase on master